test/framework/CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ add_library(integration_framework
     )
 target_link_libraries(integration_framework
     application
+    integration_framework_config_helper
     )
 
 target_include_directories(integration_framework PUBLIC ${PROJECT_SOURCE_DIR}/test)
```

Comment:
Also `integration_framework_config_helper` does not have to be linked with `logger` anymore, so `target_link_libraries` for it can be removed.